src/Microsoft.Azure.ServiceBus/Core/MessageSender.cs

```diff
@@ -5,23 +5,98 @@ namespace Microsoft.Azure.ServiceBus.Core
 {
     using System;
     using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading;
     using System.Threading.Tasks;
+    using Microsoft.Azure.Amqp;
+    using Microsoft.Azure.Amqp.Encoding;
+    using Microsoft.Azure.Amqp.Framing;
+    using Microsoft.Azure.ServiceBus.Amqp;
+    using Microsoft.Azure.ServiceBus.Primitives;
 
-    internal abstract class MessageSender : ClientEntity, IMessageSender
+    public sealed class MessageSender : ClientEntity, IMessageSender
```

Comment:
>sealed [](start = 11, length = 6)

MessageReceiver is not sealed. We should maintain consistency between the two. IMO, we should not have them sealed